src/internal/core.rs

```diff
@@ -98,12 +105,9 @@ impl<P: Package, V: Version> State<P, V> {
             // Iterate over incompatibilities in reverse order
             // to evaluate first the newest incompatibilities.
             let mut conflict_id = None;
-            for &incompat_id in self.incompatibilities.iter().rev() {
+            // We only care about incompatibilities if it contains the current package.
+            for &incompat_id in self.incompatibilities[&current_package].iter().rev() {
```

Comment:
Alright the big advantage is here!

Reply:
Yes this is where the savings happens, on the `zuse` benchmark a lot of time was spend in `current_incompat.get`. Bye keeping them indexed we can jump straight to the relevant ones without having to check them all.